No Items sync - Error with NoneType object

Hi,

i didn´t get some items in my HA-shopping list. After setting up the configuration I statarted the AddOn (on x86 installation on IntelNuc). I get this out of the protocol tab:

[13:48:21] INFO: Starting Alexa Shopping List Sync
[2024-09-02 13:48:38 +0200] [85] [INFO] Running on http://0.0.0.0:5000 (CTRL + C to quit)
[2024-09-02 13:48:38,757] [ERROR] An error occurred while fetching items: Failed to get items: {'code': 'home_assistant_error', 'message': 'Service call requested response data but did not match any entities'}
Task exception was never retrieved
future: <Task finished name='Task-1' coro=<main() done, defined at /app/main.py:271> exception=TypeError("'NoneType' object is not iterable")>
Traceback (most recent call last):
  File "/app/main.py", line 274, in main
    await update_local_shopping_list(driver, config)
  File "/app/main.py", line 101, in update_local_shopping_list
    existing_item_names = [item["summary"].lower() for item in existing_items]
                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
TypeError: 'NoneType' object is not iterable

sorry, I updated the addon and get nearly same error, different line. You need something more for debugging?


[06:39:16] INFO: Starting Alexa Shopping List Sync
[2024-09-03 06:39:23 +0200] [85] [INFO] Running on http://0.0.0.0:5000 (CTRL + C to quit)
[2024-09-03 06:39:24,669] [ERROR] An error occurred while fetching items: Failed to get items: {'code': 'home_assistant_error', 'message': 'Service call requested response data but did not match any entities'}
Task exception was never retrieved
future: <Task finished name='Task-1' coro=<main() done, defined at /app/main.py:274> exception=TypeError("'NoneType' object is not iterable")>
Traceback (most recent call last):
  File "/app/main.py", line 277, in main
    await update_local_shopping_list(driver, config)
  File "/app/main.py", line 101, in update_local_shopping_list
    existing_item_names = [item["summary"].lower() for item in existing_items]
                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
TypeError: 'NoneType' object is not iterable

--- a/alexa_shopping_list_sync/app/main.py
+++ b/alexa_shopping_list_sync/app/main.py
@@ -98,10 +98,10 @@ async def update_local_shopping_list(driver: Driver, config) -> None:
     try:
         existing_items = await ha_ws.get_todo_list_items()
 
-        existing_item_names = [item["summary"].lower() for item in existing_items]
-
         if existing_items is None:
             existing_item_names = []
+        else:
+            existing_item_names = [item["summary"].lower() for item in existing_items]
 
         for shopping_list_item in shopping_list_items:
             if shopping_list_item.lower() not in existing_item_names:

